library/src/interactor_impl.cxx

```diff
@@ -71,8 +75,101 @@ class interactor_impl::internals
 #endif
   }
 
+  //----------------------------------------------------------------------------
+  // Method defined to normalize the Z axis so all models are treated temporarily
+  // as Z-up axis models.
+  void zUpTransforms(vtkMatrix3x3* to, vtkMatrix3x3* from)
+  {
+    vtkRenderer* renderer =
+      this->VTKInteractor->GetRenderWindow()->GetRenderers()->GetFirstRenderer();
+    const double* up = renderer->GetEnvironmentUp();
+    const double* right = renderer->GetEnvironmentRight();
+    double fwd[3];
+    vtkMath::Cross(right, up, fwd);
+    const double m[9] = {
+      right[0], right[1], right[2], //
+      fwd[0], fwd[1], fwd[2],       //
+      up[0], up[1], up[2],          //
+    };
+    to->DeepCopy(m);
+    from->DeepCopy(m);
+    from->Transpose();
+  }
+
+  //----------------------------------------------------------------------------
+  // Set the view orbit position on the viewport.
+  enum class ViewType
+  {
+    VT_FRONT,
+    VT_BACK,
+    VT_RIGHT,
+    VT_LEFT,
+    VT_TOP,
+    VT_ISOMETRIC
+  };
+  static void SetViewOrbit(ViewType view, internals* self)
+  {
+    vtkNew<vtkMatrix3x3> toZup, fromZup;
+    self->zUpTransforms(toZup, fromZup);
+    camera& cam = self->Window.getCamera();
+    vector3_t up = { 0, 0, 1 };
+    point3_t pos = cam.getPosition();
+    point3_t foc = cam.getFocalPoint();
+    point3_t Axis, newPos;
+
+    /* convert coords to +Z up */
+    toZup->MultiplyPoint(pos.data(), pos.data());
+    toZup->MultiplyPoint(foc.data(), foc.data());
+
+    const double dx = foc[0] - pos[0];
+    const double dy = foc[1] - pos[1];
+    const double dz = foc[2] - pos[2];
+    double radius = sqrt(dx * dx + dy * dy + dz * dz);
+    switch (view)
+    {
+      case ViewType::VT_FRONT:
+        Axis = { 0, 0, 1 };
+        fromZup->MultiplyPoint(up.data(), up.data());
+        break;
+      case ViewType::VT_BACK:
```

Comment:
Looks like `VT_BACK` and `VT_LEFT` are not covered by the unit test. Is there a way to include these cases?